dspMetadataGUI/util/metaDataSet.py

```diff
@@ -0,0 +1,1290 @@
+"""
+This module holds the data structure that is used for modelling metadata.
+
+The classes defined here aim to represent a metadata-set, closely following the
+[metadata ontology](https://github.com/dasch-swiss/dsp-ontologies/blob/main/dsp-repository/v1/dsp-repository.shacl.ttl).
+"""
+
+from abc import ABC, abstractmethod
+import re
+from typing import Tuple
+import pyshacl
+import validators
+from rdflib import Graph, URIRef, RDF, Literal, Namespace, BNode
+from rdflib.namespace import SDO, XSD
+
+from . import utils
+from .utils import Cardinality, Datatype, Validity, IRIFactory
+
+
+ontology_url = "https://raw.githubusercontent.com/dasch-swiss/dsp-ontologies/main/dsp-repository/v1/dsp-repository.shacl.ttl"
+dsp_repo = Namespace("http://ns.dasch.swiss/repository#")
```

Comment:
Should the url [http://ns.dasch.swiss/repository#](url) be a valid link?

Reply:
no, that's just a RDF namespace to ensure the IRI is globally unique.